Use boost::none instead of the default none_t() constructor, which has
disappeared in recent boost versions.

--- include/roboptim/trajectory/spline-length.hh~	2015-06-27 08:50:04.000000000 +0200
+++ include/roboptim/trajectory/spline-length.hh	2018-04-09 11:42:29.065744689 +0200
@@ -53,7 +53,7 @@
     /// \param nDiscretizationPoints number of discretization points
     SplineLength (const CubicBSpline& spline,
 		  const size_type nDiscretizationPoints = 100,
-		  boost::optional<interval_t> interval = boost::none_t ());
+		  boost::optional<interval_t> interval = boost::none);
 
     virtual ~SplineLength ();
 
